doc/classes/CollisionObject2D.xml

```diff
@@ -216,6 +258,16 @@
 		</method>
 	</methods>
 	<members>
+		<member name="collision_layer" type="int" setter="set_collision_layer" getter="get_collision_layer" default="1">
+			The physics layers this CollisionObject2D is in.
+			Collision objects can exist in one or more of 32 different layers. See also [member collision_mask].
+			See [url=https://docs.godotengine.org/en/latest/tutorials/physics/physics_introduction.html#collision-layers-and-masks]Collision layers and masks[/url] in the documentation for more information.
+		</member>
+		<member name="collision_mask" type="int" setter="set_collision_mask" getter="get_collision_mask" default="1">
+			The physics layers this CollisionObject2D scans. See also [member collision_layer].
+			This collision object will interact with another collision object if the other collision object is in a layer that this collision object scans.
```

Comment:
This description should still state the way it currently works, as it did in `Area` and `PhysicsBody` before this PR:
`A contact is detected if object A is in any of the layers that object B scans, or object B is in any layers that object A scans.`
And that could be added to both `collision_layer` and `collision_mask` for clarity.

Once a proposal is written and agreed on, the logic for collision layers and masks can be changed and the documentation updated, but it's unrelated to this PR. Even then, it will have to state more precisely how it works, because the logic might be different for dynamic bodies, kinematic bodies and areas.
(see discussion in #42268)

Reply:
Updated.